### DISCUSSION TITLE
starting infoboard

### DISCUSSION THREAD

**Discussion by BrotiMcBrot** [SixBytesUnder/infoboard#30] [Q&A]
Hi,

I'm pretty new to github and I'm horrible at programming so sorry if my question seems to be dumb.

I've follow your tutorial but I cannot start the app? If I'm going through my browser on the pi with the ip 
It only shows me a start page of "The Vue.js Framework" and a button to get to nuxtjs.org.

Did I set something wrong? Or should I start the app from elsewhere?

Thank you for your help

Best regards
Broti

**Comment by SixBytesUnder** [comment]
Hi Broti
I haven't received any notification that you opened a discussion here.
Looks like you didn't pull the code properly, since you see the default page.
Did you follow all the steps from documentation? Did you have any issues at any point? Did you have to change anything?
What command are you running to start the app?

Can you paste here the nginx config you have and double check that you have all different files in your 'components' folder?

**Reply by BrotiMcBrot** [reply]
I'm using a **Raspberry Pi A+** for this Project and I've installed **Raspberry Pi OS (32 Bit)**.

At first i updated the Database with **sudo apt update** and sudo **apt full-upgrade**

I got the package to **/var/www/html/** with **git clone** (First two steps) changed the env.example to .env.

I didn't use tiny.vim for the next step, because for me the App is to complex. Likewise, I used nano for this step

Then I change the default file, needed to comment a few lines out with # which weren't working.

If im starting the pm2 board a message "Listening on http://localhost:3000" appears after one minute. And than?

**Reply by SixBytesUnder** [reply]
That all looks fine so far.
How does your nginx config file looks like?
Can you confirm to me the contents of your 'components' folder?

**Reply by BrotiMcBrot** [reply]
So thats what inside of the components folder is:
![2021-02-13-114846_1366x768_scrot](https://user-images.githubusercontent.com/78359766/107849499-7c9bd500-6df3-11eb-86d9-aa8d9d5dd826.png)

und hier ist die nginx config:

server {
    listen 80 default_server;
    listen [::]:80 default_server;

    access_log /var/log/nginx/infoboard-access.log;
    error_log /var/log/nginx/infoboard-error.log;

    gzip            on;
    gzip_types      text/plain application/xml text/css application/javascript;
    gzip_min_length 1000;

    # change 192.168.1.99 to your RPi's local IP address
    server_name _ 192.168.2.108 infoboard;

    location / {
        proxy_redirect                      off;
        proxy_set_header Host               $host;
        proxy_set_header X-Real-IP          $remote_addr;
        proxy_set_header X-Forwarded-For    $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto  $scheme;
        proxy_read_timeout                  1m;
        proxy_connect_timeout               1m;
        proxy_pass                          http://127.0.0.1:3000;
    }
}

**Reply by SixBytesUnder** [reply]
That all looks correct.
While on your RPi Open your browser and go to http://127.0.0.1:3000 as well as http://192.168.2.108/ and please show me both screenshots.
And then run
`tail /var/log/nginx/infoboard-access.log`
and
`tail /var/log/nginx/infoboard-error.log`
Let's see if anything gets saved there.

**Reply by SixBytesUnder** [reply]
Also I'm assuming you've actually edited the .env file and entered your own API keys and such. Not just renamed it :)

**Reply by BrotiMcBrot** [reply]
ok so ive done everything

thats the two ip adresses
![2021-02-15-181307_1366x768_scrot](https://user-images.githubusercontent.com/78359766/107981567-393f9300-6fba-11eb-973b-6a43b372e28c.png)

![2021-02-15-181416_1366x768_scrot](https://user-images.githubusercontent.com/78359766/107981576-3e9cdd80-6fba-11eb-9bfc-cf43e90d1864.png)


thats the two logs

![2021-02-15-181527_1366x768_scrot](https://user-images.githubusercontent.com/78359766/107981606-4a889f80-6fba-11eb-8483-7f61f5a46c07.png)

![2021-02-15-181635_1366x768_scrot](https://user-images.githubusercontent.com/78359766/107981614-4fe5ea00-6fba-11eb-9fc0-0ca9e020875d.png)


and thats my .env file. it is in /var/www/html/infoboard together with the env.example file

# Define where your background images should come from
## available options [local|single|nasa|unsplash|pexels|flickr]
IMAGES_SOURCE=nasa
## images directory from which to take images shown as background. Traverses all subfolders as well
IMAGES_DIR='/media/path/to/images/'
## if SINGLE_IMAGE value is "true", then the very first image found in IMAGES_DIR will be used and will never change
SINGLE_IMAGE=false
## number of seconds for how long one background image stays on before switching to next in line
IMAGE_INTERVAL=60
## toggle showing "next image" and "next folder" buttons in the bottom right corner
NAV_BUTTONS=true
## show EXIF data button
EXIF=true

# Magic Mirror mode
## Setting it to `true` will overwrite any "IMAGE" setting. Just solid black background and all white text for high contrast
MAGIC_MIRROR=false

# Auto Expand - automatically open collapsable elements
## useful if infoboard will be displayed on a non-touch screen
AE_TFL=false
AE_FORECAST=false
AE_WEATHER_DETAILS=false
AE_CALENDAR=false

# NASA Picture of the day
## apply for API key here: https://api.nasa.gov/index.html#apply-for-an-api-key
NASA_KEY=DEMO_KEY

# Unsplash photos
## register for a free account and generate your API keys https://unsplash.com/
UNSPLASH_ACCESS=
UNSPLASH_SECRET=
UNSPLASH_CALLBACK_URL=""
## show images tagged with current weather conditions
## if set to false, Unsplash will return random images
UNSPLASH_WEATHER_TAGGED=true

# Pexels photos
## register for a free account and generate your API key https://www.pexels.com/api/
PEXELS_KEY=
## search for images tagged with current weather conditions
## if set to false, Pexels will return curated images
PEXELS_WEATHER_TAGGED=true

# Flickr images
## there's no option for random image, therefore Flickr images are always weather tagged
## Create a new app and get your API key
FLICKR_API_KEY=

# Weather
WEATHER=true
## register for ClimCell free account https://www.climacell.co/weather-api/pricing/
## and generate your API key for a free version
WEATHER_API_KEY=
## latitude and longitude of location for weather forecast
WEATHER_LAT=51.507388
WEATHER_LON=-0.127768
## location name to be displayed above temperature - informational purpose only
WEATHER_LOCATION_NAME="London, UK"
## units type to display temperature in. Possible values [si] for Metric and [us] for Royal unit system
WEATHER_UNITS=si
## list of data fields for current weather. Notice, pollen data is only for USA
## full referrence: https://developer.climacell.co/v3/reference#data-layers-weather
WEATHER_FIELDS=temp,feels_like,humidity,wind_speed,wind_gust,baro_pressure,surface_shortwave_radiation,moon_phase,weather_code,pm25,pm10,o3,no2,co,so2,epa_health_concern,pollen_tree,pollen_weed,pollen_grass
## round temperature to the nearest integer
WEATHER_ROUNDED=true

# Transport for London
TFL=true
## create a free TfL account and register your app https://api.tfl.gov.uk/
TFL_APP_ID=
TFL_APP_KEY=
## comma separated list of London bus stop codes. Find your bus code here: https://tfl.gov.uk/travel-information/stations-stops-and-piers/
TFL_BUS_STOPS=490013767T,490013767T
## service status for London transport other than buses
TFL_STATUS=tube,overground,dlr,tflrail,tram

# Time and date options
## Check https://momentjs.com/docs/#/displaying/ for documentation
TIME_FORMAT="HH:mm:ss"
DATE_FORMAT="dddd, Do MMMM YYYY"

# Calendar events in iCal format
## shows up to 10 events from the next 60 days
CALENDAR_ENABLE=true
CALENDAR_DATE_FORMAT="YYYY-MM-DD"
CALENDAR_TIME_FORMAT="HH:mm"
## for Google Calendar go to settings for your calendar and
## paste below a link from "Public address in iCal format" section
CALENDAR_ICAL=""
CALENDAR_LIMIT=10

# DHT sensor for temperature and humidity
DHT=false
DHT_SENSOR_TYPE=22
DHT_GPIO_PIN=4
DHT_TIMER=1800

# SDS011 sensor for air quality
SDS011=false
# COM5 is for Windows; RPi will be ttyUSB0 or ttyUSB1...
SDS_PORT="COM5"
# How ofter to take reading from sensor (in seconds)
SDS_FREQUENCY=3600

# COVID data
C19_ENABLE=true
## list of countries to pull data for
## 3 letter country codes: https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3
C19_COUNTRY="GBR,USA"
C19_DATE_FORMAT="YYYY-MM-DD"
## data refresh frequency, default 6 hours
C19_TIMER=21600

**Reply by SixBytesUnder** [reply]
Some options from the .env file may throw errors unless you add API keys or disable them, but I assume you just removed them here for privacy reasons.
Everything else looks normal.
The only thing that I'm thinking is that you didn't run the `npm run build` command or it failed very early and the app has not been built. Try the following:

    cd /var/www/html/infoboard/
    npm install
    npm run build

Be aware that build can take a while and sometimes it fails for no reason. If it gets stuck around 90% for a minute or longer, just press `Ctrl + C` to stop the process and delete `node_modules` and `.nuxt` directories, then run `npm install` and `npm run build` again.
If you already have pm2 running completely remove previous process like this

    pm2 stop infoboard
    pm2 delete infoboard

and then, while still in `/var/www/html/infoboard/` run

    pm2 start npm --name "infoboard" -- start

**Reply by BrotiMcBrot** [reply]
ok to be honest : i didn't change anything in .env, especially for this post.

so that's happend after i used your commands:
![2021-02-16-202426_1366x768_scrot](https://user-images.githubusercontent.com/78359766/108118606-82651500-7096-11eb-93f7-2ea3c9aa66f3.png)


![2021-02-16-202614_1366x768_scrot](https://user-images.githubusercontent.com/78359766/108118421-403bd380-7096-11eb-8112-c49398d8b231.png)

**Reply by SixBytesUnder** [reply]
That's where your problem is :)
On your first screenshot you can see the green bar going up to 90% and just below it the "server" has an empty dot. And below that you have a few red "ERR" messages.
That means the build process hasn't finished. It failed at 90% of rendering client side and hasn't even touched the server side. That means your app has not been properly built and when you run it you see the basic Vue page.

OK, at least I know what's going on. The problem is that I know of this issue and can't figure out why it's happening. It happens to me sometimes as well. 

Let's fix your `.env` file first:

1. Since you have `IMAGES_SOURCE=nasa` go to `https://api.nasa.gov/index.html#apply-for-an-api-key` and register for a NASA API key. It's free.
2. Paste the key you get into `NASA_KEY=your-key-here`
3. Either generate the weather API key or change `WEATHER=true` to `WEATHER=false`
4. Change `TFL=true` to `TFL=false`
5. Since you have `CALENDAR_ENABLE=true` you have to provide a URL in `CALENDAR_ICAL=""`
6. I suggest you just turn it off for now with `CALENDAR_ENABLE=false`

Wrong options may mess up your build process. Once you fix above do:

1. Simply run `npm run build` again.
2. If it fails again, delete `.nuxt` folder and run `npm run build` again.
3. If it succeeds run `pm2 infoboard restart`

You should not see any red "ERR" messages. Instead, when the build was successful you should see a list of rendered assets, all in green.

If above still fails, try below as a last resort:

1. On your own PC (not on Raspberry), create a new folder.
2. Pull down code from github: `git clone https://github.com/SixBytesUnder/infoboard.git .`
3. Copy `.env` file from Raspberry, or just create a new one right on your PC. Just make sure your `.env` file is the same in both locations.
4. Run `npm install`
5. Run `npm run build`
6. Delete `.nuxt` folder on your Raspberry, if it exists `/var/www/html/infoboard/.nuxt/`
7. Copy the whole `.nuxt` folder from your PC to Raspberry `/var/www/html/infoboard/`
8. Leave all the other files as they are.
9. Run `pm2 restart infoboard`

**Note** - every time you make a change to `.env` file, you'll need to run `npm run build` and then `pm2 restart infoboard`

**Reply by BrotiMcBrot** [reply]
ok i did it like you said.
made everything on my pc
copied .nuxt on my Pi
and the same err came like on the last time when i built npm run

![2021-02-20-110400_1366x768_scrot](https://user-images.githubusercontent.com/78359766/108593496-e96f1c00-736b-11eb-90e8-36724b2f186b.png)

![2021-02-20-110406_1366x768_scrot](https://user-images.githubusercontent.com/78359766/108593500-f5f37480-736b-11eb-8f64-b2acd5bd3188.png)
Any idea what im doing wrong ?

**Reply by SixBytesUnder** [reply]
you don't need to build the app on your RPi. Look at the steps I wrote above. After you copy .nuxt from your PC to RPi, just run `pm2 restart infoboard`

**Reply by BrotiMcBrot** [reply]
Ok I wrote maybe too few to explain what I did.

I startet pm2 with "pm2 start npm --name "infoboard" --start
And went to my browser and it is still on the two ip addresses the starting page of "the Vue.js Framework" 

So i tried to build the app again with npm run and still have no calender

**Reply by BrotiMcBrot** [reply]
Ok I have it. I made the server again on my PC and made the whole env file new. In my last i had the nasa key wrong.

So yippie for the strike. Thank you very much. 

Now I need to find out how I change everything. 😁
